libcudacxx/include/cuda/std/__floating_point/mask.h

```diff
@@ -22,189 +22,36 @@
 #endif // no system header
 
 #include <cuda/std/__floating_point/nvfp_types.h>
+#include <cuda/std/__floating_point/properties.h>
 #include <cuda/std/__floating_point/storage.h>
-#include <cuda/std/__type_traits/always_false.h>
-#include <cuda/std/__type_traits/is_same.h>
 
 _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
-template <class _Tp>
-_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI constexpr __fp_storage_t<_Tp> __fp_sign_mask_impl() noexcept
-{
-  constexpr __fp_format __fmt = __fp_format_of_v<_Tp>;
-
-  if constexpr (__fmt == __fp_format::__binary16 || __fmt == __fp_format::__bfloat16)
-  {
-    return __fp_storage_t<_Tp>{0x8000u};
-  }
-  else if constexpr (__fmt == __fp_format::__binary32)
-  {
-    return __fp_storage_t<_Tp>{0x80000000u};
-  }
-  else if constexpr (__fmt == __fp_format::__binary64)
-  {
-    return __fp_storage_t<_Tp>{0x8000000000000000ull};
-  }
-  else if constexpr (__fmt == __fp_format::__binary128)
-  {
-    return __fp_storage_t<_Tp>{0x8000000000000000ull} << 64;
-  }
-  else if constexpr (__fmt == __fp_format::__fp80_x86)
-  {
-    return __fp_storage_t<_Tp>{0x8000ull} << 64;
-  }
-  else if constexpr (__fmt == __fp_format::__fp8_nv_e4m3 || __fmt == __fp_format::__fp8_nv_e5m2)
-  {
-    return __fp_storage_t<_Tp>{0x80u};
-  }
-  else if constexpr (__fmt == __fp_format::__fp8_nv_e8m0)
-  {
-    return __fp_storage_t<_Tp>{0x00u};
-  }
-  else if constexpr (__fmt == __fp_format::__fp6_nv_e2m3 || __fmt == __fp_format::__fp6_nv_e3m2)
-  {
-    return __fp_storage_t<_Tp>{0x20u};
-  }
-  else if constexpr (__fmt == __fp_format::__fp4_nv_e2m1)
-  {
-    return __fp_storage_t<_Tp>{0x8u};
-  }
-  else
-  {
-    static_assert(__always_false_v<_Tp>, "Unsupported floating point format");
-  }
-}
-
-template <class _Tp>
-inline constexpr __fp_storage_t<_Tp> __fp_sign_mask_v = __fp_sign_mask_impl<_Tp>();
+template <__fp_format _Fmt>
```

Comment:
I believe we should come up with a clear depiction of the layout of bits.

That will help users (aka us in 5 month) a lot when dealing with those bitfields